include/rocksdb/filter_policy.h

```diff
@@ -65,6 +70,16 @@ class FilterBitsBuilder {
   // The ownership of actual data is set to buf
   virtual Slice Finish(std::unique_ptr<const char[]>* buf) = 0;
 
+  // For internal usage
+  virtual Slice Finish(
```

Comment:
Note to @pdillinger:

I can't think of a different clean way of relaying the reservation handle for final filter (i.e, 'mutable_buf_'), other than having a new `FilterBitsBuilder::Finish` to take it.

For subclasses extending `FilterBitsBuilder` without implementing this new `FilterBitsBuilder::Finish` , when called with this new API, the call will be redirected to use its implemented `FilterBitsBuilder::Finish`, dropping the handle (e.g, `TestFilterBitsBuilder`, `CountUniqueFilterBitsBuilderWrapper ` in [full_filter_block_test.cc](https://github.com/facebook/rocksdb/blob/main/table/block_based/full_filter_block_test.cc)). However, there could be **a compiler warning `[-Werror=overloaded-virtual]`** on those subclasses which requires suppressing or adding `using FilterBitsBuilder::Finish` like I did to them. Not sure if this can be an issue.

For subclasses extending `FilterBitsBuilder` implementing this new `Finish` API (only internal classes `FastLocalBloomBitsBuilder`, `Standard128RibbonBitsBuilder` for now), they can implement the old `Finish` API to call the new API with null handle instead, like I did.